contracts/container/contract.go

```diff
@@ -277,7 +277,7 @@ func SubmitObjectPut(metaInformation []byte, sigs [][]interop.Signature) {
 		}
 	}
 	vub := getFromMap(metaMap, "validuntil").(int)
-	if vub <= ledger.CurrentIndex() {
+	if vub <= contract.Call(storage.Get(ctx, netmapContractKey).(interop.Hash160), "epoch", contract.ReadOnly).(int) {
```

Comment:
I'd still expect blocks here. Epochs are too coarse.

Reply:
I know you have problems calculating it on different nodes, but this can be solved in different ways.

Reply:
And epochs are not much different here.

Reply:
> I'd still expect blocks here. Epochs are too coarse.

Well, maybe. Still, we need to sync them somehow.

> I know you have problems calculating it on different nodes, but this can be solved in different ways.

Like the last block in the current epoch + const? I don't mind but I have nothing to inspire me here. Epoches are just a little bit more closer to our storage.

> And epochs are not much different here.

In the node it is currently creation epoch + const.

Reply:
But if it's in blocks then that's the value to use, isn't it? One node sets it, others are either OK or not with it (usually they're OK).

Reply:
My suggestion is to drop this patch.

Reply:
> My suggestion is to drop this patch.

i don't mind but what block to use in the nodes then? i can only think of https://github.com/nspcc-dev/neofs-contract/blob/f5a31fd10a31988b019feaa6a04aeaf27f71afbf/contracts/netmap/contract.go#L454 + const and hope it is not gonna tick while signatures are exchanged

Reply:
You can use current + something, it is exchanged between nodes, so it's not a problem for other nodes to use the value.